### PR TITLE
[ADD] Odoo 18.0

### PR DESCRIPTION
**Pull request by josep-tecnativa**
CC @Tecnativa TT51095

### CODE REVIEW

**Comment by ap-wtioit** [comment]
Note: the github tests seem broken for a while, https://github.com/Tecnativa/doodba/actions/runs/11072819133/job/30767952052 should have tested 16.0 from the variables but did test 17.0.

So now all matrix builds test 18.0

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -35,11 +35,14 @@ jobs:
       fail-fast: false
       matrix:
         # Test modern Odoo versions with latest Postgres version
-        odoo_version: ["17.0"]
+        odoo_version: ["18.0"]
         pg_version: ["15"]
         python_version: ["3.10"]
```
Comment (chienandalu) [inline .github/workflows/ci.yaml:40]: I'd say Odoo is supporting oficially 3.12 for this version

Comment (pedrobaeza) [reply .github/workflows/ci.yaml:40]: No, talking with them, they still stick to 3.10 although there are the corresponding changes for 3.12 compatibility

**Comment by josep-tecnativa** [comment]
> Note: the github tests seem broken for a while, https://github.com/Tecnativa/doodba/actions/runs/11072819133/job/30767952052 should have tested 16.0 from the variables but did test 17.0.
> 
> So now all matrix builds test 18.0

Fixed here https://github.com/Tecnativa/doodba/pull/626/commits/08ead1ae31bdb82ad39c001563140f312f59f2f5 . Thank you!

**Comment by pedrobaeza** [comment]
17.0 CI shouldn't fail, isn't it?

**Comment by josep-tecnativa** [comment]
@pedrobaeza ready to merge now.

**Pull request merged**